xcode/Subconscious/Shared/Components/Common/Profile/GenerativeProfilePic.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  GenerativeProfilePic.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 25/4/2023.
+//
+
+import SwiftUI
+import GameplayKit
+
+struct RandomNumberGeneratorWithSeed: RandomNumberGenerator {
+    private let randomSource: GKARC4RandomSource
+
+    init(seed: String) {
+        let data = Data(seed.utf8)
+        self.randomSource = GKARC4RandomSource(seed: data)
+        
+        // To improve the quality of the random numbers, it's recommended to drop the first N values
+        randomSource.dropValues(1024)
+    }
+
+    mutating func next() -> UInt64 {
+        let highBits = UInt64(bitPattern: Int64(randomSource.nextInt()))
+        let lowBits = UInt64(bitPattern: Int64(randomSource.nextInt()))
+        return highBits << 32 | lowBits
+    }
+}
+
+struct GenerativeProfilePicParams {
+    var sigil: String
+    var blendMode: BlendMode
+    var color: Color
+    var gradient: [Gradient.Stop]
+    
+    private static let gradientOptions = [
+        (Color.brandMarkPink, Color.brandMarkViolet, Color.brandMarkCyan),
+        (Color.brandMarkRed, Color.brandMarkPurple, Color.brandMarkViolet),
+        (Color.brandBgBlush, Color.brandMarkPurple, Color.brandMarkCyan),
+        (Color.brandMarkRed, Color.brandMarkViolet, Color.brandMarkCyan),
+        (Color.brandMarkRed, Color.brandMarkPurple, Color.brandMarkPink),
+        (Color.yellow, Color.orange, Color.brandMarkRed),
+        (Color.brandMarkPink, Color(red: 0.25, green: 0.478, blue: 1, opacity: 1), Color.brandMarkRed),
+        (Color.white, Color.brandBgBlush, Color.brandMarkPink),
+        (Color.brandMarkPink, Color(red: 0.25, green: 0.478, blue: 1, opacity: 1), Color.brandMarkCyan),
+        (Color.brandMarkCyan, Color(red: 0.25, green: 0.478, blue: 1, opacity: 1), Color.orange)
+    ]
+    
+    private static let sigilOptions = [
```

Comment:
Very cool. Eventually we may want to png-ify these for reproducibility across platforms.